GlimpseImputationPipeline/Glimpse2MergeBatches.wdl

```diff
@@ -61,17 +119,169 @@ workflow Glimpse2MergeBatches {
+    input_vcfs: {
+      localization_optional: true
+    }
+  }
+
+  command <<<
+    set -euo pipefail
+
+    # --ignore-safety-checks makes a big performance difference so we include it in our invocation.
+    # This argument disables expensive checks that the file headers contain the same set of
+    # genotyped samples and that files are in order by position of first record.
+    gatk --java-options "-Xms6000m -Xmx6500m" \
+      GatherVcfsCloud \
+      --ignore-safety-checks \
+      --gather-type BLOCK \
+      --input ~{sep=" --input " input_vcfs} \
+      --output ~{output_vcf_name}
+
+    tabix ~{output_vcf_name}
+  >>>
+
+  runtime {
+    memory: "~{machine_mem_mb} MiB"
+    cpu: "1"
+    bootDiskSizeGb: 15
+    disks: "local-disk " + disk_size_gb + " HDD"
+    preemptible: 1
+    docker: gatk_docker
+  }
+
+  output {
+    File output_vcf = "~{output_vcf_name}"
+    File output_vcf_index = "~{output_vcf_name}.tbi"
+  }
+}
+
+
+task SubsetToIntervals {
+    input {
+        File vcf
+        File vcf_index
+        File interval_list
+        Int disk_size_gb = ceil(size(vcf, "GiB") + 50)
+
+        String docker_gatk
+    }
+
+    parameter_meta {
+        vcf: {
+                localization_optional: true
+            }
+
+        vcf_index: {
+                localization_optional: true
+            }
+    }
+
+    command <<<
+        gatk SelectVariants -V ~{vcf} -L ~{interval_list} -O subset_intervals.vcf.gz
+    >>>
+
+    output {
+        File subset_vcf = "subset_intervals.vcf.gz"
+        File subset_vcf_index = "subset_intervals.vcf.gz.tbi"
+    }
+
+    runtime {
+        docker: docker_gatk
+        memory: "2000 MiB"
+        disks: "local-disk " + disk_size_gb + " HDD"
     }
 }
 
+task CountVariants {
+    input {
+        File vcf
+        String docker_gatk
+    }
+
+    parameter_meta {
+        vcf: {
+                localization_optional: true
+            }
+    }
+
+    command <<<
+        gatk CountVariants -V ~{vcf} -O count.txt
+    >>>
+
+    output {
+        Int count = read_int("count.txt")
+    }
+
+    runtime {
+        docker: docker_gatk
+    }
+
+}
+
+task ScatterIntervalList {
+  input {
+    File interval_list
+    Int scatter_count
+  }
+
+  command <<<
+    set -e
+    mkdir out
+    java -Xms1000m -Xmx1500m -jar /usr/gitc/picard.jar \
+      IntervalListTools \
+      SCATTER_COUNT=~{scatter_count} \
+      SUBDIVISION_MODE=INTERVAL_SUBDIVISION \
+      UNIQUE=true \
+      SORT=true \
+      INPUT=~{interval_list} \
+      OUTPUT=out
+
+    python3 <<CODE
+    import glob, os
+    # Works around a JES limitation where multiples files with the same name overwrite each other when globbed
+    intervals = sorted(glob.glob("out/*/*.interval_list"))
+    for i, interval in enumerate(intervals):
+      (directory, filename) = os.path.split(interval)
+      newName = os.path.join(directory, str(i + 1).zfill(4) + filename)
+      os.rename(interval, newName)
+    print(len(intervals))
```

Comment:
Works perfectly fine, it just feels a little nicer if it would be written to an actual file, don't you think?
```suggestion
    with open('num_intervals.txt', 'w') as num_intervals:
      num_intervals.write(f'{len(intervals)}\n')
```

Reply:
yup, agreed